apollo-router/src/executable.rs

```diff
@@ -327,12 +327,16 @@ pub fn main() -> Result<()> {
 
     let mut builder = tokio::runtime::Builder::new_multi_thread();
     builder.enable_all();
-    if let Some(nb) = std::env::var("APOLLO_ROUTER_NUM_CORES")
+
+    // This environment variable is intentionally undocumented.
+    // See also APOLLO_ROUTER_COMPUTE_THREADS in apollo-router/src/compute_job.rs
+    if let Some(nb) = std::env::var("APOLLO_ROUTER_IO_THREADS")
         .ok()
         .and_then(|value| value.parse::<usize>().ok())
     {
         builder.worker_threads(nb);
```

Comment:
Worth logging these details as well:
```
tracing::info!(size=nb, type="io", source="env", "thread pool");
```

Reply:
Can `tracing` be used before we create a Tokio runtime? (If not, I can move things around to log after we’re in async context)

Reply:
When I added this log it was captured in the snapshot for test `pq_layer_freeform_graphql_with_safelist_log_unknown_true` which made the outcome of the test dependent of number of available CPUs. Rather than spend time on figuring out how to make that test filter out that specific log I’ve removed the log for now.